src/main/java/com/datasonnet/XMLFormatPlugin.java

```diff
@@ -65,6 +67,23 @@ public Value read(String inputXML, Map<String, Object> params) throws Exception
 
     public String write(Value inputXML, Map<String, Object> params) throws Exception {
         JSONObject input = new JSONObject(UjsonUtil.jsonObjectValueTo(inputXML));
+
+        if (params.containsKey(ROOT_ELEMENT)) {
```

Comment:
I'm not going to hold up the PR but these ifs would look lovely inside a method returning the root instead of mutating `input` directly